pkg/api/v1/publish-api-types.go

```diff
@@ -0,0 +1,138 @@
+//
+// Keys: Required and must have length >= 1 and <= 21 (`maxKeysPerPublish`)
+//
+// HealthAuthorityID: The identifier for the mobile application.
+//  - Android: The App Package AppPackageName
+//  - iOS: The BundleID
+//
+// VerificationPayload: The Verification Certificate from a verification server.
+// hmacKey: the device generated secret that is used to recalculate the HMAC value
+//  that is present in the verification payload.
+//
+// SymptomOnsetInterval: An interval number that aligns with the symptom onset date.
+//  - Uses the same interval system as TEK timing.
+//  - Will be rounded down to the start of the UTC day provided.
+//  - Will be used to calculate the days +/- symptom onset for provided keys.
+//  - MUST be no more than 14 days ago.
+//  - Does not have to be within range of any of the provided keys (i.e. future
+//    key uploads)
+//
+// Traveler - set to true if the TEKs in this publish set are consider to be the
+//  keys of a "traveler" who has left the home region represented by this server
+//  (or by the home health authority in case of a multi-tenant installation).
+//
+// RevisionToken: An opaque string that must be passed intact on additional
+//   publish requests from the same device, where the same TEKs may be published
+//   again.
+//
+// VisitedCountries: list (possibly empty) where the device has travelled to.
+//
+// ReportType: type of report - is it self-report, confirmed diagnose, ...?
+//
+// ConsentToFederation: whether user of the device has allowed to share his keys to other countries.
+//
+// Padding: random base64 encoded data to obscure the request size. The server will
+// not process this data in any way.
+//
+// Partial success: If at least one of the Keys passed in is valid, then the publish
+// request will accept those keys, return a response code of 200 (OK) AND also
+// return a 'Code' of ErrorPartialFailure along with an error message of
+// exactly which keys were not accepted and why. This does not indicate a failure
+// that must be reported to the user, but does indicate an issue with the application
+// making the upload (sending invalid data).
+type PublishKeysRequestDevice struct {
+	Keys                 []DeviceExposureKey `json:"temporaryExposureKeys" validate:"required"`
+	HealthAuthorityID    string              `json:"healthAuthorityID" validate:"required"`
+	VerificationPayload  string              `json:"verificationPayload"`
+	HMACKey              string              `json:"hmacKey"`
+	SymptomOnsetInterval int32               `json:"symptomOnsetInterval"`
+	Traveler             bool                `json:"traveler"`
+	RevisionToken        string              `json:"revisionToken" validate:"required"`
+
+	VisitedCountries    []string   `json:"visitedCountries"`
+	ReportType          ReportType `json:"reportType"`
+	ConsentToFederation bool       `json:"consentToFederation"`
+
+	Padding string `json:"padding"`
+}
```

Comment:
this has the same problem as before

why not
```
type PublishKeysRequestDevice struct {
         keyserverapi.Publish //embedded struct

	VisitedCountries    []string   `json:"visitedCountries"`
	ReportType          ReportType `json:"reportType"`
	ConsentToFederation bool       `json:"consentToFederation"`
}
```

Reply:
Oh. Didn't know about this syntax :smile: